openslides_backend/action/actions/committee/import_meeting.py

```diff
@@ -0,0 +1,241 @@
+
+        for collection in shall_be_empty:
+            if meeting_json.get(collection):
+                raise ActionException(f"{collection} must be empty.")
+
+        for user in meeting_json.get("user", []):
+            if not user["password"] == "":
+                raise ActionException("User password must be an empty string.")
+
+        self.check_usernames_and_generate_new_ones(meeting_json)
+        self.update_committee_id(instance)
+        self.generate_random_passwords(meeting_json)
+        self.set_enable_anonymous(meeting_json)
+        self.set_imported_at(meeting_json)
+
+        # replace ids in the meeting_json
+        replace_map = self.create_replace_map(meeting_json)
+        self.replace_fields(instance, replace_map)
+        return instance
+
+    def check_usernames_and_generate_new_ones(self, json_data: Dict[str, Any]) -> None:
+        used_usernames = set()
+        for entry in json_data["user"]:
+            username_unique = False
+            template_username = entry["username"].rstrip("0123456789")
+            count = 1
+            while not username_unique:
+                if entry["username"] in used_usernames:
+                    entry["username"] = template_username + str(count)
+                    count += 1
+                    continue
+                result = self.datastore.filter(
+                    Collection("user"),
+                    FilterOperator("username", "=", entry["username"]),
+                    ["id"],
+                )
+                if result:
+                    entry["username"] = template_username + str(count)
+                    count += 1
+                    continue
+                username_unique = True
+            used_usernames.add(entry["username"])
+
+    def update_committee_id(self, instance: Dict[str, Any]) -> None:
+        json_data = instance["meeting_json"]
+        json_data["meeting"][0]["committee_id"] = instance["id"]
+
+    def generate_random_passwords(self, json_data: Dict[str, Any]) -> None:
+        for entry in json_data["user"]:
+            entry["password"] = get_random_string(10)
+
+    def set_enable_anonymous(self, json_data: Dict[str, Any]) -> None:
+        json_data["meeting"][0]["enable_anonymous"] = False
+
+    def set_imported_at(self, json_data: Dict[str, Any]) -> None:
+        json_data["meeting"][0]["imported_at"] = round(time.time())
+
+    def create_replace_map(
+        self, json_data: Dict[str, Any]
+    ) -> Dict[str, Dict[int, int]]:
+        replace_map: Dict[str, Dict[int, int]] = {}
+        for collection in json_data:
+            replace_map[collection] = {}
+            new_ids = self.datastore.reserve_ids(
+                Collection(collection), len(json_data[collection])
+            )
+            for entry, new_id in zip(json_data[collection], new_ids):
+                replace_map[collection][entry["id"]] = new_id
+        return replace_map
```

Comment:
store this as a member of this class, so you do not have to pass it around.

Reply:
Done.